tls-aggregation.tf

```diff
@@ -11,24 +11,24 @@
 # Files: tls/{aggregation-ca.crt,aggregation-ca.key}
 
 resource "tls_private_key" "aggregation-ca" {
-  count = "${var.enable_aggregation == "true" ? 1 : 0}"
+  count = var.enable_aggregation == true ? 1 : 0
 
   algorithm = "RSA"
   rsa_bits  = "2048"
 }
 
 resource "tls_self_signed_cert" "aggregation-ca" {
-  count = "${var.enable_aggregation == "true" ? 1 : 0}"
+  count = var.enable_aggregation == true ? 1 : 0
 
-  key_algorithm   = "${tls_private_key.aggregation-ca.algorithm}"
-  private_key_pem = "${tls_private_key.aggregation-ca.private_key_pem}"
+  key_algorithm   = tls_private_key.aggregation-ca[0].algorithm
+  private_key_pem = tls_private_key.aggregation-ca[0].private_key_pem
```

Comment:
oh, had to think about this one but was easy: terraform 0.12 returns an array if count is used, no matter if `count=1` is used. Therefore we need to use `[0]`.

I'll mark this as solved, just to make sure if in the future we thing we missed something, this was done on purpose :)